package.json

```diff
@@ -54,25 +54,22 @@
     "karma-jasmine": "~4.0.0",
     "chromium": "^3.0.3",
     "puppeteer": "^15.5.0",
-    "ng-packagr": "^14.0.2",
+    "ng-packagr": "^15.2.2",
     "protractor": "~7.0.0",
     "rxjs": "~6.6.0",
     "ts-node": "~8.3.0",
-    "tslib": "^2.0.0",
-    "typescript": "~4.7.4",
+    "tslib": "^2.3.0",
+    "typescript": "~4.9.5",
     "zone.js": "~0.11.4"
   },
   "dependencies": {
     "@microsoft/applicationinsights-shims": "^2.0.2",
     "@microsoft/applicationinsights-common": "^2.8.12",
     "@microsoft/applicationinsights-core-js": "^2.8.12",
-    "@microsoft/applicationinsights-properties-js": "^2.8.12",
-    "tslib": "^2.0.0"
```

Comment:
I'll need to go back over why we added this, but from memory someone had issues because this was missing.
I'm guessing because of the defaults for TypeScript about importing these helper functions.

Reply:
I ain't sure if it's required to be duplicated since `tslib` is already in dev dependencies.

Reply:
Dev dependencies are not required for anyone using this component, while dependencies are always as specified and peer dependencies tell the build environment that they need to have at lease the matching version.

So have in dev dependencies but not in peer dependencies or dependencies (could) cause anyone consuming this package to end up with an error. If they are using TypeScript then it's not generally an issue (unless they use the tsconfig to cause the helper to be directly injected), but there are some how are just consuming this as a JavaScript component (they don't have TypeScript) and that would again cause an issue.

Reply:
ie. Dev dependencies are required for building this component only and not for consumers of the package.

Reply:
I do understand the difference between those, but when you run `npm run build` it outputs built assets to `dist/LIB_NAME` which has its own `package.json`. Aren't you publishing from the dist folder?

Reply:
Yes, I believe that the dist is published as part of the package, but my understanding is that (at least for the previous version that I was more directly involved with) the package.json is derived from the this one, ie. the dependencies are derived (copied) from this one as part of the build process.

Reply:
Nope, it copies the library package.json:
```ts
const packageJson = { ...entryPoint.packageJson };
```

Checks if `tslib` is defined or not, adds it to `dependencies` if not defined:
```ts
if (!packageJson.peerDependencies?.tslib && !packageJson.dependencies?.tslib) {
  const tsLibVersion = angularPeerDependencies.tslib || angularDependencies.tslib;

  if (tsLibVersion) {
    packageJson.dependencies = {
      ...packageJson.dependencies,
      tslib: tsLibVersion,
    };
  }
}
```

Then removes these properties from the package.json object:

```ts
const packageJsonPropertiesToDelete = [
    'stylelint',
    'prettier',
    'browserslist',
    'devDependencies',
    'jest',
    'workspaces',
    'husky',
  ];
```

Writes the file:
```ts
await writeFile(path.join(entryPoint.destinationPath, 'package.json'), JSON.stringify(packageJson, undefined, 2));
```

Reply:
Good to know